File: chain-sdk/clojure/src/on_chain_api/specs/uniswap_v3_position_snapshot_v3_dto.clj
(ns on-chain-api.specs.uniswap-v3-position-snapshot-v3-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def uniswap-v3-position-snapshot-v3-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :id) string?
   (ds/opt :owner) string?
   (ds/opt :pool) string?
   (ds/opt :position) string?
   (ds/opt :timestamp) string?
   (ds/opt :liquidity) string?
   (ds/opt :deposited_token_0) string?
   (ds/opt :deposited_token_1) string?
   (ds/opt :withdrawn_token_0) string?
   (ds/opt :withdrawn_token_1) string?
   (ds/opt :collected_fees_token_0) string?
   (ds/opt :collected_fees_token_1) string?
   (ds/opt :transaction) string?
   (ds/opt :fee_growth_inside_0_last_x128) string?
   (ds/opt :fee_growth_inside_1_last_x128) string?
   (ds/opt :vid) int?
   })

(def uniswap-v3-position-snapshot-v3-dto-spec
  (ds/spec
    {:name ::uniswap-v3-position-snapshot-v3-dto
     :spec uniswap-v3-position-snapshot-v3-dto-data}))

File: chain-sdk/javascript-closure-angular/API/Client/UniswapV3PositionSnapshotV3DTO.js
goog.provide('API.Client.UniswapV3.PositionSnapshotV3DTO');

/**
 * @record
 */
API.Client.UniswapV3PositionSnapshotV3DTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.recvTime;

/**
 * Number of block in which entity was recorded.
 * @type {!number}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.blockNumber;

/**
 * NFT token identifier, format: <NFT token id>#<block number>
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.id;

/**
 * Owner of the NFT.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.owner;

/**
 * Pool the position is within.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.pool;

/**
 * Position of which the snap was taken of.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.position;

/**
 * Timestamp of block in which the snap was created.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.timestamp;

/**
 * Total position liquidity.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.liquidity;

/**
 * Amount of token 0 ever deposited to position.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.depositedToken0;

/**
 * Amount of token 1 ever deposited to position.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.depositedToken1;

/**
 * Amount of token 0 ever withdrawn from position (without fees).
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.withdrawnToken0;

/**
 * Amount of token 1 ever withdrawn from position (without fees).
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.withdrawnToken1;

/**
 * All time collected fees in token0.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.collectedFeesToken0;

/**
 * All time collected fees in token1.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.collectedFeesToken1;

/**
 * Transaction in which the snapshot was initialized.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.transaction;

/**
 * Variable needed for fee computation.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.feeGrowthInside0LastX128;

/**
 * Variable needed for fee computation.
 * @type {!string}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.feeGrowthInside1LastX128;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.UniswapV3PositionSnapshotV3DTO.prototype.vid;

